### DISCUSSION TITLE
Ist this possible: Width of p:dialog depending on zoom factor of browser.

### DISCUSSION THREAD

**Discussion by hgfrey** [primefaces/community#783] [PrimeFaces] [answered]
Is it possible to change the width of p:dialog depending on the zoom factor for the browser?

**Comment by melloware** [comment]
Zoom Factor? So you don't want `responsive=true` ?

> responsive | true | Boolean | In responsive mode the dialog adjusts itself based on screen width, browser resizing, and scrolling of the document.

There is also this you would have to subscribe to an event and change the width with JS: https://stackoverflow.com/questions/995914/catch-browsers-zoom-event-in-javascript

**Reply by hgfrey** [reply]
How does responsive interact with other attributes? Like width for example.

**Reply by hgfrey** [reply]
Accessibility is the key word. People with disabilities must be able to use the application. This requires that the application can still be operated when I zoom to 300% on a Full HD monitor. To do this, the elements must be placed one below the other at some point.

**Comment by hgfrey** [comment]
I made the dialog responsive, but the result is not what I wanted. The entire width of the monitor is always used. This doesn't look nice and at a zoom level of 300% the dialog can no longer be used.

            <p:dialog
                id="dialog"
                widgetVar="dateiAnfuegenDialogWidgetAllgemeineInformationen"
                header="Datei anfügen"
                draggable="true"
                resizable="true"
                closable="false"
                modal="true"
                responsive="true">

100%:
![image](https://github.com/primefaces/community/assets/33621098/ef459020-ecf5-44d5-8a89-f26d494b0b42)

300%:
![image](https://github.com/primefaces/community/assets/33621098/7093875f-1d5f-48ec-89a4-786d33792f4e)

**Reply by hgfrey** [reply, marked as answer]
Setting fitViewport to true solves the 300% problem.

**Reply by melloware** [reply]
Yep `fitViewPort` is what you want.
